docs: refactor installation page (#8487)
langflow-ai/langflow: docs/docs/Get-Started/get-started-installation.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> Langflow can be installed in multiple ways.

Reviewer: ```suggestion
Langflow can be installed in multiple ways:
```

> * **Recommended option**: Download and install the [standalone desktop application](#install-and-run-langflow-desktop) for the easiest setup experience.

Reviewer: Is there a difference between the contents of LF Desktop, the Docker image, and the Python package? For example, does the Python package include extra dependencies or something that facilitate deveopment? It might be good to indicate those differences here. For example, if I cannot create custom components with LF Desktop, I might want to know that upfront.

* **Docker**: Pull and run the [Docker image](#install-and-run-langflow-docker) to start a Langflow container.

* **For developers**: Install the [Langflow OSS Python package](#install-and-run-langflow-oss).

## Install and run Langflow Desktop

**Langflow Desktop** is a desktop version of Langflow that includes all the features of open source Langflow, with an additional [version management](#manage-your-langflow-version-in-langflow-desktop) feature for managing your Langflow version.

<Tabs groupId="os">
  <TabItem value="macOS" label="macOS">

  1. Navigate to [Langflow Desktop](https://www.langflow.org/desktop).
  2. Enter your **Name**, **Email address**, and **Company**, and then click **Download**.

>   3. Open the **Finder**, and then navigate to **Downloads**.
>   4. Double-click the downloaded `.dmg` file.
>   5. To install Langflow Desktop, drag and drop the application icon to the **Applications** folder.

Reviewer: Do we need to be this detailed about basic steps to install a program? 

```suggestion
  3. Mount and install the Langflow application.
```

  6. When the installation completes, open the Langflow application.

  </TabItem>
  <TabItem value="Windows" label="Windows">

  1. Navigate to [Langflow Desktop](https://www.langflow.org/desktop).
  2. Enter your **Name**, **Email address**, and **Company**, and then click **Download**.
  3. Open the **File Explorer**, and then navigate to **Downloads**.
  4. Double-click the downloaded `.msi` file.
  5. To install Langflow Desktop, follow the instructions to install the executable file.
  6. When the installation completes, open the Langflow application.

  </TabItem>

</Tabs>

> The application checks [uv](https://docs.astral.sh/uv/concepts/tools/), your local environment, and the Langflow version, and then starts.

Reviewer: You mention uv here but don't state that it's required to install LF Desktop. Do they need to have uv installed or will the LF Desktop command install it as a dependency?

Author: LF Desktop installs it

### Manage your Langflow version in Langflow Desktop

When a new version of Langflow is available, Langflow Desktop displays an upgrade message.

To manage your Langflow version in Langflow Desktop, follow these steps:

1. To access Langflow Desktop's **Version Management** pane, click your **Profile Image**, and then select **Version Management**.
Langflow Desktop's current version is displayed, with other version options listed after it.
The **latest** version is always highlighted.
2. To change your Langflow version, select another version.
A confirmation pane containing the selected version's changelog appears.
3. To change to the selected version, click **Confirm**.
The application restarts with the new version installed.

## Install and run Langflow with Docker {#install-and-run-langflow-docker}

[Docker](https://docs.docker.com/) provides the simplest way to run Langflow in an isolated environment to ensure consistent behavior across different systems and eliminate dependency conflicts.
The Langflow Docker image is available on [Docker Hub](https://hub.docker.com/r/langflowai/langflow).

> 1. With Docker installed and running on your system, run this command to pull the `latest` Langflow image and start it:

Reviewer 2: ```suggestion
1. With Docker installed and running on your system, run the following command to pull the `latest` Langflow image and start it:
```

    ```bash
    docker run -p 7860:7860 langflowai/langflow:latest
    ```

2. To start using Langflow, visit `http://localhost:7860/`.

For more information, see [Deploy Langflow on Docker](/deployment-docker).

## Install and run Langflow OSS

Before you install and run Langflow OSS, be sure you have the following items.

- [Python 3.10 to 3.13](https://www.python.org/downloads/release/python-3100/)
- [uv](https://docs.astral.sh/uv/getting-started/installation/)
- A dual-core CPU and at least 2 GB of RAM. More intensive use requires a multi-core CPU and at least 4 GB of RAM.
- A virtual environment created with [uv](https://docs.astral.sh/uv/pip/environments)

<details>
<summary>Need help with virtual environments?</summary>

Virtual environments ensure Langflow is installed in an isolated, fresh environment.

> To create a new virtual environment, do the following.

Author: Added

<Tabs groupId="os">
  <TabItem value="macOS" label="macOS">
    1. Navigate to where you want your virtual environment to be created, and create it with `uv`.
Replace `VENV_NAME` with your preferred name for your virtual environment.
```
uv venv VENV_NAME
```
2. Start the virtual environment.
```
source VENV_NAME/bin/activate
```
Your shell's prompt changes to display that you're currently working in a virtual environment.
```
(VENV_NAME) ➜  langflow git:(main) ✗
```
3. To deactivate the virtual environment and return to your regular shell, type `deactivate`.
To delete the virtual environment, type `rm -rf VENV_NAME`.

>   </TabItem>

Reviewer 2: Take a look at your tab structure. Instructions for both OS types appear on each tab.

Replace `VENV_NAME` with your preferred name for your virtual environment.
```
uv venv VENV_NAME
```
2. Start the virtual environment.
```shell
VENV_NAME\Scripts\activate
```
Your shell's prompt changes to display that you're currently working in a virtual environment.
```
(VENV_NAME) PS C:/users/username/langflow-dir>
```
3. To deactivate the virtual environment and return to your regular shell, type `deactivate`.
To delete the virtual environment, type `Remove-Item VENV_NAME`.
  </TabItem>
  </Tabs>
  <TabItem value="Linux" label="Linux">
    1. Navigate to where you want your virtual environment to be created, and create it with `uv`.
Replace `VENV_NAME` with your preferred name for your virtual environment.
```
uv venv VENV_NAME
```
2. Start the virtual environment.
```
source VENV_NAME/bin/activate
```
Your shell's prompt changes to display that you're currently working in a virtual environment.
```
(VENV_NAME) ➜  langflow git:(main) ✗
```
3. To deactivate the virtual environment and return to your regular shell, type `deactivate`.
To delete the virtual environment, type `rm -rf VENV_NAME`.
  </TabItem>
</details>

<Tabs groupId="os">

>   <TabItem value="macOS" label="macOS">

Reviewer: Actually, these should be included with common install issues. If you decide to make the troubleshooting page like I suggested, can you please move those there? Or move these down to the common install issues section.

Then add a sentence to step 1 like:

```
If you encounter an error during installation, such as `C++ Build Tools Required!`, see [troubleshooting link].
```

    1. To install Langflow, run the following command.
    ```bash

>     uv pip install langflow

Reviewer 2: Can you adjust the indentation on the commands so that they left-align below the start of the text instead of to the left of the number?

    ```

  </TabItem>
  <TabItem value="Windows" label="Windows">
  :::important
  Windows installations of Langflow OSS require a C++ compiler, such as [Visual Studio](https://visualstudio.microsoft.com/downloads/), that may not be present on your system. If you receive a `C++ Build Tools Required!` error, follow the on-screen prompt to install Microsoft C++ Build Tools, or visit the Visual Studio download link above.
  :::

      1. To install Langflow, run the following command.
      ```bash
      uv pip install langflow
      ```

>   :::tip

Reviewer 2: Having this tip admonition so close to the important admonition above makes the screen look a little cluttered. Suggest removing the tip and presenting the information as regular text instead.

  If you experience an error from the `webrtcvad` package, run `uv pip install webrtcvad-wheels` in the virtual environment, and then try installing again.
  :::

  </TabItem>
  <TabItem value="Linux" label="Linux">

      1. To install Langflow, run the following command.
      ```bash
      uv pip install langflow
      ```

  </TabItem>

</Tabs>

> 2. After installation, to run Langflow, use the following command:

Reviewer: I also suggesting adding here than it takes some time for LF to start, and there is _no_ feedback in the terminal until it has started (at which point you get the welcome message).

```bash
uv run langflow run
```

3. To confirm that a local Langflow instance starts, go to the default Langflow URL at `http://127.0.0.1:7860`.

> After confirming that Langflow is running, create your first flow with the [Quickstart](/get-started-quickstart).

Reviewer 2: Suggest adding a link to the quickstart at the end of the instructions for installing Langflow Desktop, too.

> ### Manage Langflow OSS versions

Reviewer 2: This is a grab-bag of miscellaneous commands. I suggest the following changes:

- Make upgrading its own topic.
- Move the info about installing a specific version to the OSS installation topic. 
- Make reinstalling its own topic and add some context around why/when a reinstallation may be necessary.

Reviewer: I will add this section to the upgrade topic task in our project plan.

To upgrade Langflow to the latest version, use the following command:

```bash
uv pip install langflow -U
```

To install a specific version of the Langflow package, add the required version to the command:

```bash
uv pip install langflow==1.3.2
```

To reinstall Langflow and all of its dependencies, add the `--force-reinstall` flag to the command:

```bash
uv pip install langflow --force-reinstall
```

> ### Install optional dependencies for Langflow OSS

Reviewer: IMO, this extra dependency stuff should all go somewhere else, like the install-custom-dependencies topic (Under "Develop") section, and the topic should be renamed to "Manage Langflow OSS dependencies".

Then, this section should link to the dependency management page, similar to the Install Docker section that points to the docker deployment page for more information.

If this isn't possible with the current release schedule, I would suggest putting these extra things in `<details>` so the Install page looks at minimal as possible and promotes only the minimum required steps.

> ### Install optional dependencies for Langflow OSS

Author: Moved under "details"

Langflow OSS provides optional dependency groups that extend its functionality.

These dependencies are listed in the [pyproject.toml](https://github.com/langflow-ai/langflow/blob/main/pyproject.toml#L191) file under `[project.optional-dependencies]`.

Install dependency groups using pip's `[extras]` syntax. For example, to install Langflow with the `postgresql` dependency group, enter the following command:

```bash
uv pip install "langflow[postgresql]"
```

To install multiple extras, enter the following command:

```bash
uv pip install "langflow[deploy,local,postgresql]"
```

To add your own custom dependencies, see [Install custom dependencies](/install-custom-dependencies).

> ### Stop Langflow OSS

Author: Removed

    :::

  1. Navigate to [Langflow Desktop](https://www.langflow.org/desktop).
  2. Enter your **Name**, **Email address**, and **Company**, and then click **Download**.
  3. Open the **File Explorer**, and then navigate to **Downloads**.
  4. Double-click the downloaded `.msi` file.
  5. To install Langflow Desktop, follow the instructions to install the executable file.
  6. When the installation completes, open the Langflow application.

  After confirming that Langflow is running, create your first flow with the [Quickstart](/get-started-quickstart).

  </TabItem>

</Tabs>

### Manage your version of Langflow Desktop

When a new version of Langflow is available, Langflow Desktop displays an upgrade message.

To manage your version of Langflow Desktop, follow these steps:

  1. In Langflow Desktop, click your profile image, and then select **Version Management**.
  The *Version Management* pane lists your active Langflow version first, followed by other available versions.
  The **latest** version is always highlighted.
  2. To change your Langflow version, select another version.

>   A confirmation pane containing the selected version's changelog appears.

Reviewer: Should we say something here like "If the selected version contains potential breaking changes for your flows, you might want to export/backup your flows before changing the version" ? I'm not sure what would actually be recommended here, aside from a fresh LF Desktop install in a fresh venv, and then recreating the flows to test them on the new version.

>   A confirmation pane containing the selected version's changelog appears.

Author: Added

>   3. To change to the selected version, click **Confirm**.
>   The application restarts with the new version installed.

Reviewer: ```suggestion
  3. To apply the change, click **Confirm**.
  Langflow desktop restarts to install and activate the new version.
```

## Install and run Langflow with Docker {#install-and-run-langflow-docker}

You can use the [Langflow Docker image](https://hub.docker.com/r/langflowai/langflow) to run Langflow in an isolated environment.
Running applications in [Docker](https://docs.docker.com/) containers ensures consistent behavior across different systems and eliminates dependency conflicts.

1. Install and start [Docker](https://docs.docker.com/).
2. Pull the latest [Langflow Docker image](https://hub.docker.com/r/langflowai/langflow) and start it:

  ```bash
  docker run -p 7860:7860 langflowai/langflow:latest
  ```

3. To access Langflow, navigate to `http://localhost:7860/`.

For more information, see [Deploy Langflow on Docker](/deployment-docker).

> ## Install and run Langflow OSS

Reviewer: ```suggestion
## Install and run the Langflow OSS Python package
```

Should we specify that this is the Python package variant, since they are all technically Langflow OSS?

```
source VENV_NAME/bin/activate
```
Your shell's prompt changes to display that you're currently working in a virtual environment.
```
(VENV_NAME) ➜  langflow git:(main) ✗
```
3. To deactivate the virtual environment and return to your regular shell, type `deactivate`.
To delete the virtual environment, type `rm -rf VENV_NAME`.
  </TabItem>
  <TabItem value="Windows" label="Windows">
1. Navigate to where you want your virtual environment to be created, and create it with `uv`.
Replace `VENV_NAME` with your preferred name for your virtual environment.
```
uv venv VENV_NAME
```
2. Start the virtual environment.
```shell
VENV_NAME\Scripts\activate
```
Your shell's prompt changes to display that you're currently working in a virtual environment.
```
(VENV_NAME) PS C:/users/username/langflow-dir>
```
3. To deactivate the virtual environment and return to your regular shell, type `deactivate`.
To delete the virtual environment, type `Remove-Item VENV_NAME`.
  </TabItem>
  </Tabs>

</details>

2. To install Langflow, run the following command.
  ```bash
  uv pip install langflow
  ```

3. After installation, start Langflow:
  ```bash
  uv run langflow run
  ```

> 4. To confirm that a local Langflow instance is running, navigate to the default Langflow URL`http://127.0.0.1:7860`.

Reviewer: ```suggestion
4. To confirm that a local Langflow instance is running, navigate to the default Langflow URL `http://127.0.0.1:7860`.
It can take a few minutes for Langflow to start.
```

After confirming that Langflow is running, create your first flow with the [Quickstart](/get-started-quickstart).

### Manage Langflow OSS versions

To manage your Langflow version, use the following commands.

<details closed>

> <summary>Install dependency groups</summary>

Reviewer: ```suggestion
<summary>Manage Langflow OSS versions</summary>
```
